test/helpers.test.js

```diff
@@ -0,0 +1,220 @@
+'use strict';
+
+const { Helpers } = require('../src/helpers');
+
+const { expect } = require('chai');
+const sinon = require('sinon');
+
+describe('Helper functions', () => {
+	describe('hasProperties', () => {
+		it('Returns true when it has only the properties passed', () => {
+			const obj = {
+				field1: 'test1',
+				field2: 'test2'
+			};
+
+			expect(Helpers.hasProperties(obj, ['field1', 'field2'])).to.be.true;
+		});
+
+		it('Returns true when it has the properties passed and some other', () => {
+			const obj = {
+				field1: 'test1',
+				field2: 'test2',
+				field3: 'test3'
+			};
+
+			expect(Helpers.hasProperties(obj, ['field1', 'field2'])).to.be.true;
+		});
+
+		it('Returns true when no properties checked', () => {
+			const obj = {
+				field1: 'test1',
+				field2: 'test2'
+			};
+
+			expect(Helpers.hasProperties(obj, [])).to.be.true;
+		});
+
+		it('Returns false when it is missing some properties', () => {
+			const obj = {
+				field1: 'test1',
+				field2: 'test2'
+			};
+
+			expect(Helpers.hasProperties(obj, ['field1', 'field2', 'field3'])).to.be.false;
+		});
+
+		it('Returns false when there are none of the properties', () => {
+			const obj = {
+				fake1: 'test1',
+				fake2: 'test2'
+			};
+
+			expect(Helpers.hasProperties(obj, ['field1', 'field2'])).to.be.false;
+		});
+	});
+});
+
+describe('Extract JSON', () => {
+	it('Able to extract when there is only a JSON', () => {
+		const json = {
+			field1: 'test1',
+			field2: 'test2'
+		};
+
+		const text = JSON.stringify(json);
+
+		expect(Helpers.extractJSON(text)).to.deep.equal(json);
+	});
+
+	it('Able to extract when there is JSON and text', () => {
+		const json = {
+			field1: 'test1',
+			field2: 'test2'
+		};
+
+		const text = `More text${JSON.stringify(json)}More text`;
+
+		expect(Helpers.extractJSON(text)).to.deep.equal(json);
+	});
+
+	it('Able to extract when there is multiple JSONs', () => {
+		const json = {
+			field1: 'test1',
+			field2: 'test2'
+		};
+
+		const ignoredJson = {
+			fake1: 'test1',
+			field2: 'test2'
+		};
+
+		const text = `${JSON.stringify(ignoredJson)}${JSON.stringify(json)}`;
+
+		expect(Helpers.extractJSON(text, ['field1', 'field2'])).to.deep.equal(json);
+	});
+
+	it('Returns null when there is no json', () => {
+		const text = 'Just text';
+
+		expect(Helpers.extractJSON(text)).to.deep.equal(null);

```

Comment:
Nit: probably `to.equal(null)` makes a little more sense than `to.deep.equal(null)`, since there's nothing "deep" to check

Reply:
I actually need the `deep.equal`. If I do the normal equal the equality fails even if they are equal

Reply:
my bad, I misread the code, you are correct I do not need the deep equal in this case